### ISSUE TITLE
Create a Contributing file

### ISSUE THREAD

**Issue by bonimy** [Ersanio/SMAS-Disassembly#6]
# Issue
The project needs a thorough outline that details how contributions can best be submitted.

# Features
> A CONTRIBUTING file tells your audience how to participate in your project.
> -[Open Source Guides](https://opensource.guide/starting-a-project/#writing-your-contributing-guidelines)

# Resolution
I'm suggesting the following resources to guide in designing a CONTRIBUTING file
* [Writing your contributing guidelines](https://opensource.guide/starting-a-project/#writing-your-contributing-guidelines)
* [Contributing Template](https://github.com/nayafia/contributing-template/blob/master/CONTRIBUTING-template.md)
* [Mozilla Contributing Guidelines](http://mozillascience.github.io/working-open-workshop/contributing/)

The CONTRIBUTING file should have the following sections
## Warm Welcome
Thank the user for wanting to contribute to the project, and state the importance of getting to know the contributing guidelines
### How to file bug reports
We should probably make a template for this. Important things to add would be
* How to reproduce the bug
* A stack trace if available
* Screenshots if applicable
### How to suggest a new feature
I'm not too sure where to take this one immediately. An issue template could help. One of the biggest problems is like "You should add support for X" where is very large and no extra information is given. So I guess a suggestion should be as detailed as the feature is large.
### How to set up your environment and run tests
Provide a thorough guide on how to get started, successfully build the code, and run all tests to make sure they have a properly configured environment.
### Code styles
A set of enforceable code styles to avoid conflicting formatting styles from separate contributors.
### The types of contributions we're looking for.
Basically. We want to keep it within the scope of what the project actually does. Adding a VISION file could be a good idea.
### Vision for the project
Give a brief description of what the project is for.
# Task List
- [ ] - Warm Welcome
- [ ] - How to file bug reports
- [ ] - How to suggest a new feature
- [ ] - How to set up your environment and run tests
- [ ] - Code styles
- [ ] - The types of contributions we're looking for
- [ ] - Vision for the project
